### PR TITLE
Enable Dynamic shader resolution

### PR DESCRIPTION
**Pull request by romulogcerqueira**
* shader resolution now is dependent of number of bins
* number of beams as dynamic property
* multibeam sonar as data-driven task
* rename shader initialization method
* correct time in output frame

### CODE REVIEW

File: tasks/MultibeamSonarTask.cpp [hunk 1]
```diff
@@ -44,8 +44,8 @@ bool MultibeamSonarTask::startHook() {
 		return false;
 
     // generate shader world
-    uint width = 3328;
-    Task::initShader(width, false);
+    uint width = sonar_sim.bin_count * 5.12;  // 5.12 pixels are needed for each bin
```
Comment (doudou) [inline tasks/MultibeamSonarTask.cpp]: `5.12` ? Why ?

File: tasks/MultibeamSonarTask.cpp [hunk 2]
```diff
@@ -91,3 +91,23 @@ void MultibeamSonarTask::stopHook() {
...
 }
+
+bool MultibeamSonarTask::setBin_count(int value) {
+    if (value <= 0) {
+        RTT::log(RTT::Error) << "The number of bins must be positive." << RTT::endlog();
+        return false;
+    }
+    sonar_sim.bin_count = value;
+    float width = sonar_sim.bin_count * 5.12;  // 5.12 pixels are needed for each bin
+    Task::setupShader(width, false);
```
Comment (doudou) [inline tasks/MultibeamSonarTask.cpp]: Don't you want to round the width upwards instead of downwards ?

File: tasks/MultibeamSonarTask.cpp [hunk 3]
```diff
@@ -91,3 +91,23 @@ void MultibeamSonarTask::stopHook() {
...
+    return (imaging_sonar_simulation::TaskBase::setBin_count(value));
+}
+
+bool MultibeamSonarTask::setBeam_count(int value) {
+    if (value < 64 || value > 512) {
+        RTT::log(RTT::Error) << "The number of beams must be between 64 and 512." << RTT::endlog();
+        return false;
+    }
+    sonar_sim.beam_count = value;
+    return (imaging_sonar_simulation::TaskBase::setBin_count(value));
```
Comment (doudou) [inline tasks/MultibeamSonarTask.cpp]: Wrong copy/paste ... should probably add unit tests.

File: tasks/MultibeamSonarTask.cpp (hunk 2 above)
Comment (doudou) [inline tasks/MultibeamSonarTask.cpp]: Next question: why is the bin-direction the width in multibeam, but the height in scanning ?

File: tasks/MultibeamSonarTask.cpp (hunk 3 above)
Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: My mistake. Done!

File: tasks/MultibeamSonarTask.cpp (hunk 1 above)
Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: I changed the shader image resolution to check the amount of pixels needed to fill all bins and avoid the "black holes" and our linear interpolation. I found this relation between the number of bins and shader resolution.

File: tasks/MultibeamSonarTask.cpp (hunk 2 above)
Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: > Don't you want to round the width upwards instead of downwards ?

Good. Done!

Comment (romulogcerqueira) [inline tasks/MultibeamSonarTask.cpp]: Initially due multibeam sonar has higher FOV-X and bigger beam count than scanning one. Do you have a better suggestion?

**Pull request merged**